src/vcpkg/base/downloads.cpp

```diff
@@ -16,23 +15,64 @@
 #include <vcpkg/base/system.proxy.h>
 #include <vcpkg/base/util.h>
 
+#include <vcpkg/commands.version.h>
+
 #include <set>
 
 using namespace vcpkg;
 
 namespace
 {
-    void set_common_curl_easy_options(CurlEasyHandle& easy_handle, StringView url, const CurlHeaders& request_headers)
+    constexpr StringLiteral vcpkg_curl_user_agent_header =
+        "User-Agent: vcpkg/" VCPKG_BASE_VERSION_AS_STRING "-" VCPKG_VERSION_AS_STRING " (curl)";
+
+    void add_curl_headers(Command& cmd, View<std::string> headers)
     {
-        auto* curl = easy_handle.get();
-        curl_easy_setopt(curl, CURLOPT_USERAGENT, vcpkg_curl_user_agent);
-        curl_easy_setopt(curl, CURLOPT_URL, url_encode_spaces(url).c_str());
-        curl_easy_setopt(curl,
-                         CURLOPT_FOLLOWLOCATION,
-                         2L); // Follow redirects, change request method based on HTTP response code.
-                              // https://curl.se/libcurl/c/CURLOPT_FOLLOWLOCATION.html#CURLFOLLOWOBEYCODE
-        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, request_headers.get());
-        curl_easy_setopt(curl, CURLOPT_HEADEROPT, CURLHEADER_SEPARATE); // don't send headers to proxy CONNECT
+        cmd.string_arg("-H").string_arg(vcpkg_curl_user_agent_header);
+        for (auto&& header : headers)
+        {
+            cmd.string_arg("-H").string_arg(header);
+        }
+    }
+
+    // Extracts the host part from a URL string.
+    std::string extract_host(StringView url)
```

Comment:
This thing needs unit tests.

Reply:
Thanks for the review. I’ll revisit this once the curl library integration has been re-landed.